### PR TITLE
Clarify use of mounts for merge patches

### PR DESCRIPTION
**Pull request by mattcary**
Fixes: #2799 

I've tested out the example patch for enabling the cache mutation detector. I haven't tested that the same strategy can be used on a worker as stated in my change, but it seems reasonable.

### CODE REVIEW

File: site/content/docs/user/configuration.md [hunk 1]
```diff
@@ -455,5 +455,47 @@ nodes:
...
+    containerPath: /patches
+
+kubeadmConfigPatches:
+  - |
+    kind: InitConfiguration
+    patches:
+      directory: /patches
+{{< codeFromInline >}}
+
+Note the `extraMounts` stanza. `kind` runs kubeadm from a container in the node,
```
Comment (BenTheElder) [inline site/content/docs/user/configuration.md:492]: nit: I think this is slightly mis-worded / confusing. For kind the node is a container, and we run kubeadm in the node-container, but not in a container "within" the node (like say, kube-proxy).

**Comment by BenTheElder** [comment]
thanks for writing this up. one nit. 

sorry for the delay, especially fun week:
https://github.com/kubernetes-sigs/kind/issues/3124
https://kubernetes.io/blog/2023/03/10/image-registry-redirect/
🙃

File: site/content/docs/user/configuration.md (hunk 1 above)
Comment (mattcary) [reply site/content/docs/user/configuration.md]: Good point thanks. Here'a another try.

**Comment by mattcary** [comment]
> sorry for the delay, especially fun week: #3124 https://kubernetes.io/blog/2023/03/10/image-registry-redirect/ 🙃

No problem, I knew responses would be delayed. If I don't get these PRs started though I forget about them...

**Pull request merged**